barman/cli.py

```diff
@@ -52,6 +54,127 @@
+            aliases = kwargs.pop('aliases', [])
+            parser = super(AliasedSubParsersAction, self).add_parser(name, **kwargs)
+
+            # Make the aliases work.
+            for alias in aliases:
+                self._name_parser_map[alias] = parser
+            # Make the help text reflect them, first removing old help entry.
+            if 'help' in kwargs:
+                help = kwargs.pop('help')
+                self._choices_actions.pop()
+                pseudo_action = self._AliasedPseudoAction(name, aliases, help)
+                self._choices_actions.append(pseudo_action)
+
+            return parser
+
+    # override argparse to register new subparser action by default
+    _ActionsContainer.__init__ = AliasedSubParsersAction._containerInit
+
+p = ArgumentParser(epilog="Barman by EnterpriseDB (www.enterprisedb.com)")
+p.add_argument(
+    "-v",
+    "--version",
+    action="version",
+    version="%s\n\nBarman by EnterpriseDB (www.enterprisedb.com)"
+    % barman.__version__,
+)
+p.add_argument(
+    "-c",
+    "--config",
+    help="uses a configuration file "
+    "(defaults: %s)" % ", ".join(barman.config.Config.CONFIG_FILES),
+    default=SUPPRESS,
+)
+p.add_argument(
+    "--color",
+    "--colour",
+    help="Whether to use colors in the output",
+    choices=["never", "always", "auto"],
+    default="auto",
+)
+p.add_argument(
+"--log-level",
+    help="Override the default log level",
+    choices=list(get_log_levels()),
+    default=SUPPRESS,
+)
+p.add_argument("-q", "--quiet", help="be quiet", action="store_true")
+p.add_argument("-d", "--debug", help="debug output", action="store_true")
+p.add_argument(
+       "-f",
+       "--format",
+       help="output format",
+       choices=output.AVAILABLE_WRITERS.keys(),
+       default=output.DEFAULT_WRITER,
+   )
+
+subparsers = p.add_subparsers(dest="command")
+
+def argument(*name_or_flags, **kwargs):
+    """Convenience function to properly format arguments to pass to the
+      command decorator.
+    """
+
+    # Remove the completer keyword argument from the dictionary
+    completer = kwargs.pop('completer', None)
+    return (list(name_or_flags), completer, kwargs)
+
+def command(args=[], parent=subparsers, cmd_aliases=[]):
+    """Decorator to define a new subcommand in a sanity-preserving way.
+    The function will be stored in the ``func`` variable when the parser
+    parses arguments so that it can be called directly like so::
+        args = cli.parse_args()
+        args.func(args)
+    Usage example::
+        @command([argument("-d", help="Enable debug mode", action="store_true")])
+        def command(args):
+            print(args)
+    Then on the command line::
+        $ python cli.py command -d
+    """
+    def decorator(func):
+        parser = parent.add_parser(func.__name__.replace('_','-'), description=func.__doc__,
+                                   help=func.__doc__, aliases=cmd_aliases)
```

Comment:
I spent time on  add the help section yesterday but I realized you just add it :) Nice job.